src/rshim.c

```diff
@@ -2157,30 +2211,155 @@ static int rshim_update_locked_mode(rshim_backend_t *bd)
       int rt;
 
       rshim_lock();
-      pthread_mutex_lock(&bd->mutex);
+      has_bd_lock = !pthread_mutex_trylock(&bd->mutex);
       rt = rshim_access_check(bd); 
-      pthread_mutex_unlock(&bd->mutex);
-      rshim_unlock();
+      if (has_bd_lock)
+        rshim_unlock();
```

Comment:
need fix here, missing pthread_mutex_unlock and changed rshim_unlock incorrectly.

Reply:
Thanks. Fixed in e36d992.